tensorflow/core/util/tensor_bundle/byte_swap.cc

```diff
@@ -102,17 +107,59 @@ Status ByteSwapTensor(Tensor* t) {
     case DT_VARIANT:
       return errors::Unimplemented(
           "Byte-swapping not yet implemented for tensors with dtype ",
-          t->dtype());
+          dtype);
 
     // Byte-swapping shouldn't make sense for other dtypes.
     default:
       return errors::Unimplemented(
-          "Byte-swapping not supported for tensors with dtype ", t->dtype());
+          "Byte-swapping not supported for tensors with dtype ", dtype);
   }
 
-  char* backing_buffer = const_cast<char*>((t->tensor_data().data()));
+  char* backing_buffer = buff;
   TF_RETURN_IF_ERROR(ByteSwapArray(backing_buffer, bytes_per_elem, array_len));
   return Status::OK();
 }
 
+Status ByteSwapTensor(Tensor* t) {
+  char* buff = const_cast<char*>((t->tensor_data().data()));
+  return ByteSwapBuffer(buff, t->tensor_data().size(), t->dtype(), t->NumElements());
+}
+
+Status SwapTensorContent(MetaGraphDef* meta_graph_def) {
+  GraphDef graph_def = *meta_graph_def->mutable_graph_def();
+  for (auto& function : *meta_graph_def->mutable_graph_def()->mutable_library()->mutable_function()) {
+    for (auto& node : (*function.mutable_node_def())) {
+      if (node.op() == "Const") {
+        auto node_iterator = node.mutable_attr()->find("value");
+        if (node_iterator != node.mutable_attr()->end()) {
+          AttrValue node_value = node_iterator->second;
+          if (node_value.has_tensor()) {
+            auto tsize = node_value.mutable_tensor()->tensor_content().size();
+            auto p_type = node_value.mutable_tensor()->dtype();
+            // Swap only when there is something in tensor_content field
```

Comment:
I think this comment still makes sense since the swapping will happen only when `tsize != 0`.

The else branch will be run only when `tsize != 0` as well, it matches `if (parsed.tensor_data().size() != 0)`.

Reply:
My bad, I was matching the else to the outer if